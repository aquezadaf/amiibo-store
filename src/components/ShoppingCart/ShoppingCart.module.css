.container {
    display: flex;
    justify-content: space-between;
}

.text {
    color: #777777;
    cursor: pointer;
}

.text:hover {
    color: black;
}

.badge {
    background-color: #f1816e;
    border-radius: 10px;
    color: white;
    display: inline-block;
    font-size: 12px;
    line-height: 1;
    padding: 3px 7px;
    text-align: center;
    vertical-align: middle;
    white-space: nowrap;
    margin-left: 5px;
    cursor: pointer;
}