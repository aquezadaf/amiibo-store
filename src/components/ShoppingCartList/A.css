.container {
  position: absolute;
  top: 50px;
  right: 10px;
}

.shopping-cart {
  margin: 20px 0;
  float: right;
  background: #faf4f4;
  width: 320px;
  position: relative;
  border-radius: 3px;
  padding: 20px;
  box-shadow: 0px 5px 31px -1px rgba(0, 0, 0, 0.15);
}

/* 
.shopping-cart-items {
  padding-top: 20px;
}

.shopping-cart-items li {
  margin-bottom: 18px;
}

.shopping-cart-items img {
  float: left;
  margin-right: 12px;
}

.shopping-cart-items .item-name {
  display: block;
  padding-top: 10px;
  font-size: 16px;
}

.shopping-cart .shopping-cart-items .item-price {
  color: #6394F8;
  margin-right: 8px;
}

.shopping-cart .shopping-cart-items .item-quantity {
  color: #ABB0BE;
}
*/

.shopping-cart:after {
  bottom: 100%;
  left: 95%;
  border: solid transparent;
  content: " ";
  height: 0;
  width: 0;
  position: absolute;
  pointer-events: none;
  border-bottom-color: #faf4f4;
  border-width: 8px;
  margin-left: -8px;
}

/*
.button {
  background: #6394F8;
  color: white;
  text-align: center;
  padding: 12px;
  text-decoration: none;
  display: block;
  border-radius: 3px;
  font-size: 16px;
  margin: 25px 0 15px 0;
}

.button:hover {
  background: #729ef9;
}

.clearfix:after {
  content: "";
  display: table;
  clear: both;
}
*/
