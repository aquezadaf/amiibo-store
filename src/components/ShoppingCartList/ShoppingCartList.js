import React from 'react';
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next';
import styles from './ShoppingCartList.module.css';

function ShoppingCartList(props) {
    const { products, onIncrementProduct, onSubtractProduct } = props
    const { t } = useTranslation();
    return (
        <div className={styles.container}>
            <div className={styles.shoppingCart}>
                {products.map(product => (
                    <div key={product.id} className={styles.item}>
                        <span>
                            <div className={styles.name}>{product.name}</div>
                            <div className={styles.quantity}>${product.price} x {product.quantity}</div>
                        </span>
                        <span>
                            <button
                                className={styles.editButton}
                                onClick={() => onIncrementProduct(product)}>{t('shoppingCart.add')}</button>
                            <button
                                className={styles.editButton}
                                onClick={() => onSubtractProduct(product)}>{t('shoppingCart.subtract')}</button>
                        </span>
                    </div>
                ))}
                <div className={styles.totalContainer}>
                    <span className={styles.total}>{t('shoppingCart.total')}</span>
                    <span className={styles.total}>$ {products.reduce((total, { quantity, price }) => total + (quantity * price), 0)}</span>
                </div>
                <a href="#" className={styles.button}>{t('shoppingCart.checkout')}</a>
            </div>
        </div>
    );
}

ShoppingCartList.propTypes = {
    products: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        quantity: PropTypes.number.isRequired
    })),
    onIncrementProduct: PropTypes.func.isRequired,
    onSubtractProduct: PropTypes.func.isRequired
}

export default ShoppingCartList
