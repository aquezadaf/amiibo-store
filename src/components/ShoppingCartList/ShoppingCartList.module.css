.container {
    position: absolute;
    top: 50px;
    right: 10px;
}

.shoppingCart {
    margin: 20px 0;
    float: right;
    background: #faf4f4;
    min-width: 320px;
    position: relative;
    border-radius: 3px;
    padding: 20px;
    box-shadow: 0px 5px 31px -1px rgba(0, 0, 0, 0.15);
}

.shoppingCart:after {
    bottom: 100%;
    left: 95%;
    border: solid transparent;
    content: " ";
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
    border-bottom-color: #faf4f4;
    border-width: 8px;
    margin-left: -8px;
}

.item {
    display: flex;
    justify-content: space-between;
}

.name {
    font-size: 15px;
    color: #393c45;
    font-weight: 700;
    text-align: left;
}

.quantity {
    font-size: 15px;
    color: #b3b3b3;
    padding: 2px 0 5px 0;
    text-align: left;
}

.price {
    color: #393c45;
    font-size: 20px;
    font-weight: 600;
    text-align: left;
    padding-bottom: 8px;
}

.editButton {
    font-size: 15px;
    color: #fff;
    border: 2px solid #fff;
    background-color: #0382b4;
    padding: 0 10px;
    cursor: pointer;
}

.editButton:hover {
    background: #029bd8;
}

.totalContainer {
    display: flex;
    justify-content: space-between;
}

.total {
    font-size: 15px;
    color: #393c45;
    font-weight: 700;
}

.button {
    display: block;
    margin: 25px 0 0 0;
    text-decoration: none;
    font-size: 15px;
    color: #fff;
    border: 2px solid #fff;
    background-color: #0382b4;
    text-align: center;
    text-transform: uppercase;
    font-weight: 700;
    padding: 8px 0;
}

.button:hover {
    background: #029bd8;
}