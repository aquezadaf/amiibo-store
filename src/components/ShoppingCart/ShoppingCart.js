import React from 'react';
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next';
import ShoppingCartList from '../../containers/ShoppingCartList';
import styles from './ShoppingCart.module.css';

function ShoppingCart(props) {
    const { count } = props
    const { t } = useTranslation();
    return (
        <div className={styles.container}>
            <span className={styles.text}>{t('navigationBar.cart')}</span>
            { count && <span className={styles.badge}>{count}</span> }
            <ShoppingCartList />
        </div>
    );
}

ShoppingCart.propTypes = {
    count: PropTypes.number,
}

export default ShoppingCart
