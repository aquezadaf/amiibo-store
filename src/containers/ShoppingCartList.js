import { connect } from 'react-redux'
import ShoppingCartList from '../components/ShoppingCartList'
import selectors from '../selectors'
import actions from '../actions'

const mapStateToProps = state => ({
    products: selectors.shoppingCart.productsSelector(state)
});

const mapDispatchToProps = dispatch => ({
    onIncrementProduct: (product) =>  dispatch(actions.shoppingCart.addProductToCart(product)),
    onSubtractProduct: (product) =>  dispatch(actions.shoppingCart.subtractProductFromCart(product))
});

export default connect(mapStateToProps, mapDispatchToProps)(ShoppingCartList);
