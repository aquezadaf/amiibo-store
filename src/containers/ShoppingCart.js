import { connect } from "react-redux";
import ShoppingCart from "../components/ShoppingCart";
import selectors from "../selectors";

const mapStateToProps = (state) => ({
  count: selectors.shoppingCart.productsSelector(state).length,
});

export default connect(mapStateToProps, null)(ShoppingCart);
