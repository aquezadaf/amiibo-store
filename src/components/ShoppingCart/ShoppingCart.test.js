import React from "react";
import { shallow } from "enzyme";
import ShoppingCart from "./ShoppingCart";
import ShoppingCartList from "../../containers/ShoppingCartList";

jest.mock("../../containers/ShoppingCartList", () => () => null);

it("renders shopping cart", () => {
  const wrapper = shallow(<ShoppingCart count={1} />);
  expect(wrapper.html()).toMatchSnapshot();
});

it("renders shopping cart list when pressing shopping cart", () => {
  const wrapper = shallow(<ShoppingCart count={1} />);
  wrapper.children().simulate("click");
  const cartList = wrapper.find(ShoppingCartList);
  expect(cartList).toHaveLength(1);
});

it("does not render shopping cart list when pressing shopping cart and having no products", () => {
  const wrapper = shallow(<ShoppingCart count={0} />);
  wrapper.children().simulate("click");
  const cartList = wrapper.find(ShoppingCartList);
  expect(cartList).toHaveLength(0);
});
