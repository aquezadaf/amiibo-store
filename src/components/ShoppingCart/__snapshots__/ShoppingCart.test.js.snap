// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders shopping cart 1`] = `"<div><div class=\\"container\\"><span class=\\"text\\"></span><span class=\\"badge\\">1</span></div></div>"`;
